ui/page/dcrdex/market.go

```diff
@@ -0,0 +1,1063 @@
+
+	// TODO: Fetch orders or order history.
+}
+
+func assetIcon(th *cryptomaterial.Theme, assetType libutils.AssetType) *cryptomaterial.Image {
+	switch assetType {
+	case libutils.DCRWalletAsset:
+		return th.Icons.DCR
+	case libutils.BTCWalletAsset:
+		return th.Icons.BTC
+	case libutils.LTCWalletAsset:
+		return th.Icons.LTC
+	}
+	return nil
+}
+
+func (pg *DEXMarketPage) marketDropdownListItem(baseAsset, quoteAsset libutils.AssetType) func(gtx C) D {
+	baseIcon, quoteIcon := assetIcon(pg.Theme, baseAsset), assetIcon(pg.Theme, quoteAsset)
+	return func(gtx cryptomaterial.C) cryptomaterial.D {
+		return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+			layout.Rigid(func(gtx C) D {
+				return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						if baseIcon == nil {
+							return D{}
+						}
+						return baseIcon.Layout20dp(gtx)
+					}),
+					layout.Rigid(func(gtx C) D {
+						return layout.Inset{Right: u2, Left: u2}.Layout(gtx, pg.Theme.Label(values.TextSize16, baseAsset.String()).Layout)
+					}),
+				)
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layout.Inset{Right: u2, Left: u2}.Layout(gtx, pg.Theme.Label(values.TextSize16, "/").Layout)
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						if quoteIcon == nil {
+							return D{}
+						}
+						return quoteIcon.Layout20dp(gtx)
+					}),
+					layout.Rigid(func(gtx C) D {
+						return layout.Inset{Right: u2, Left: u2}.Layout(gtx, pg.Theme.Label(values.TextSize16, quoteAsset.String()).Layout)
+					}),
+				)
+			}),
+		)
+	}
+}
+
+// OnNavigatedFrom is called when the page is about to be removed from
+// the displayed window. This method should ideally be used to disable
+// features that are irrelevant when the page is NOT displayed.
+// NOTE: The page may be re-displayed on the app's window, in which case
+// OnNavigatedTo() will be called again. This method should not destroy UI
+// components unless they'll be recreated in the OnNavigatedTo() method.
+// Part of the load.Page interface.
+func (pg *DEXMarketPage) OnNavigatedFrom() {}
+
+// Layout draws the page UI components into the provided C
+// to be eventually drawn on screen.
+// Part of the load.Page interface.
+func (pg *DEXMarketPage) Layout(gtx C) D {
+	return pg.pageContentLayout(gtx)
+}
+
+func (pg *DEXMarketPage) pageContentLayout(gtx C) D {
+	pageContent := []func(gtx C) D{
+		pg.priceAndVolumeDetail(),
+		func(gtx C) D {
+			return cryptomaterial.LinearLayout{
+				Width:       cryptomaterial.MatchParent,
+				Height:      cryptomaterial.WrapContent,
+				Orientation: layout.Horizontal,
+			}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					return layout.W.Layout(gtx, pg.orderForm())
+				}),
+				layout.Flexed(1, func(gtx C) D {
+					return layout.E.Layout(gtx, pg.orderbook())
+				}),
+			)
+		},
+		pg.openOrdersAndHistory,
```

Comment:
Can you be consistent in writing functions priceAndVolumeDetail  and openOrdersAndHistory ?
one return func(gtx C) D and other return D

Reply:
Okay.